api/uniter/unit_test.go

```diff
@@ -239,6 +239,35 @@ func (s *unitSuite) TestWatch(c *gc.C) {
 	wc.AssertOneChange()
 }
 
+func (s *unitSuite) TestWatchRelations(c *gc.C) {
```

Comment:
Maybe this test needs updating - it was copied across when the watching was done at an app level. Maybe there should be a scenario where we ensure the new watcher ignores unwanted relation changes.